lib/spack/spack/mixins.py

```diff
@@ -190,6 +190,8 @@ def _filter_compiler_wrappers_impl(self):
         ]
         for env_var, compiler_path in replacements:
             if env_var in os.environ:
+                p = os.path.realpath(os.environ[env_var])
+                x.filter(p, compiler_path, **filter_kwargs)
```

Comment:
This is a little tricky to read, it probably needs a comment and maybe a bit of refactoring. Maybe

```suggestion
                # filter spack wrapper and links to spack wrapper in case 
                # build system runs realpath
                wrapper = os.environ[env_var]
                for wrapper_path in (wrapper, os.path.realpath(wrapper):
                    x.filter(wrapper_path, compiler_path, **filter_kwargs)
```